lte/gateway/python/magma/pipelined/tests/test_he.py

```diff
@@ -0,0 +1,283 @@
+
+        cls.thread = start_ryu_app_thread(test_setup)
+        cls.he_controller = he_controller_reference.result()
+        cls.testing_controller = testing_controller_reference.result()
+
+    def _wait_for_responses(self, chan, response_count, logger):
+        def fail(err):
+            logger.error("Failed to install rule for subscriber: %s", err)
+        for _ in range(response_count):
+            try:
+                result = chan.get()
+
+            except MsgChannel.Timeout:
+                return fail("No response from OVS policy mixin")
+            if not result.ok():
+                return fail(result.exception())
+
+    @classmethod
+    def tearDownClass(cls):
+        stop_ryu_app_thread(cls.thread)
+        BridgeTools.destroy_bridge(cls.BRIDGE)
+
+    def tearDown(self):
+        cls = self.__class__
+        dp = HeTableTest.he_controller._datapath
+        cls.he_controller.delete_all_flows(dp)
+
+    def test_default_flows(self):
+        """
+        Verify that a proxy flows are setup
+        """
+
+        snapshot_verifier = SnapshotVerifier(self,
+                                             self.BRIDGE,
+                                             self.service_manager,
+                                             max_sleep_time=20,
+                                             datapath=HeTableTest.he_controller._datapath)
+
+        with snapshot_verifier:
+            pass
+
+    def test_ue_flows_add(self):
+        """
+        Verify that a proxy flows are setup
+        """
+        cls = self.__class__
+        self._msg_hub = MessageHub(HeTableTest.he_controller.logger)
+
+        ue_ip = '1.1.1.1'
+        dest_server = '2.2.2.2'
+        flow_msg = cls.he_controller.get_subscriber_flows(ue_ip, dest_server, 123,
+                                                          ['abc.com'], 'IMSI01', b'1')
+        chan = self._msg_hub.send(flow_msg,
+                                  HeTableTest.he_controller._datapath,)
+        self._wait_for_responses(chan, len(flow_msg), HeTableTest.he_controller.logger)
+
+        snapshot_verifier = SnapshotVerifier(self,
+                                             self.BRIDGE,
+                                             self.service_manager,
+                                             max_sleep_time=20,
+                                             datapath=HeTableTest.he_controller._datapath)
+
+        with snapshot_verifier:
+            pass
+
+    def test_ue_flows_add2(self):
+        """
+        Verify that a proxy flows are setup
+        """
+        cls = self.__class__
+        self._msg_hub = MessageHub(HeTableTest.he_controller.logger)
+        dp = HeTableTest.he_controller._datapath
+        ue_ip1 = '1.1.1.200'
+        dest_server1 = '2.2.2.4'
+        rule1 = 123
+        flow_msg = cls.he_controller.get_subscriber_flows(ue_ip1, dest_server1, rule1,
+                                                          ['abc.com'], 'IMSI01', b'1')
```

Comment:
can you rewrite this test so that it uses the policyrules functions that call he_controller.get_subscriber_flows? 
Or just add another `test_ue_flows_enf(self)`, that way we can catch more issues than just calling controller directly

Reply:
This test is just to validate the flows.
I will add separate test in enforcement test as another PR.